Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -473,15 +482,36 @@ public final class HTTPDecoder<In, Out>: ByteToMessageDecoder, HTTPDecoderDelega
     // the actual state
     private let parser: BetterHTTPParser
     private let leftOverBytesStrategy: RemoveAfterUpgradeStrategy
+    private let informalResponseStrategy: InformalResponseStrategy
     private let kind: HTTPDecoderKind
     private var stopParsing = false // set on upgrade or HTTP version error
+    private var lastHeaderWasInformal = false
 
     /// Creates a new instance of `HTTPDecoder`.
     ///
     /// - parameters:
     ///     - leftOverBytesStrategy: The strategy to use when removing the decoder from the pipeline and an upgrade was,
     ///                              detected. Note that this does not affect what happens on EOF.
-    public init(leftOverBytesStrategy: RemoveAfterUpgradeStrategy = .dropBytes) {
+    public convenience init(leftOverBytesStrategy: RemoveAfterUpgradeStrategy = .dropBytes) {
+        self.init(leftOverBytesStrategy: leftOverBytesStrategy, informalResponseStrategy: .drop)
+    }
+    
+    /// Strategy to use when a HTTPDecoder receives an informal HTTP response (1xx except 101)
+    public enum InformalResponseStrategy {
+        /// Drop the informal response and only forward the "real" response
+        case drop
+        /// Forward the informal response and then forward the "real" response
```

Comment:
We should clarify what this looks like (namely, multiple `.head` messages for a single `.end`).